.github/workflows/rust_release.yml

```diff
@@ -0,0 +1,44 @@
+name: Release to Cargo
+
+on:
+  push:
+    tags: [ 'rust-v*' ]
+
+defaults:
+  run:
+    working-directory: ./rust
+
+jobs:
+  validate-release-tag:
+    name: Validate git tag
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: compare git tag with cargo metadata
+        run: |
+          PUSHED_TAG=${GITHUB_REF##*/}
+          CURR_VER=$( grep version Cargo.toml | head -n 1 | awk '{print $3}' | tr -d '"' )
+          if [[ "${PUSHED_TAG}" != "rust-v${CURR_VER}" ]]; then
+            echo "Cargo metadata has version set to ${CURR_VER}, but got pushed tag ${PUSHED_TAG}."
+            exit 1
+          fi
+
+  release-crate:
+    needs: validate-release-tag
+    name: Release crate
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: publish
+          args: --manifest-path ./rust/Cargo.toml --token ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
would be better to inject the token using `CARGO_REGISTRY_TOKEN` instead. I don't know if github action would be smart enough to hide the token from output for us.

Reply:
I did try it on my fork and it GitHub does hide secrets from the logs:
https://github.com/FrancisMurillo/delta-rs/runs/2652351497?check_suite_focus=true#step:4:1

Reply:
Ah okay... `cargo publish` specifies CARGO_REGISTRY_TOKEN as an option. Removed `--token` option in favor of env